app/utility/second_parser.py

```diff
@@ -0,0 +1,35 @@
+def parse_duration(sec: int) -> str:
+    if sec < 0:
+        raise ValueError("Given value cannot be less than 0!")
+    if sec == 0:
+        return "0s"
+    days = sec // 86400
+    hours = (sec - days * 86400) // 3600
+    minutes = (sec - days * 86400 - hours * 3600) // 60
+    seconds = sec - days * 86400 - hours * 3600 - minutes * 60
+    result = (
+        ("{}d ".format(int(days)) if days else "")
+        + ("{}h ".format(int(hours)) if hours else "")
+        + ("{}m ".format(int(minutes)) if minutes else "")
+        + ("{}s ".format(int(seconds)) if seconds else "")
+    )
+    if (
+        result == days
+        or hours
+        or minutes
+        or seconds
+        or days
+        and not hours
+        or days
+        and not minutes
+        and days
+        and not seconds
+        or hours
+        and not minutes
+        or hours
+        and not seconds
+        or minutes
+        and not seconds
+    ):
```

Comment:
Hmm this `if` statement is hard to read and I think there is a nicer solution for this. You can remove it and  always return `result[:-1]` and make sure other cases will work with this.

Reply:
Yes, it doesn't look so nice, but simple `result[:-1]` without `if` statement is not working as it should. I don't have any other idea how to make it works without modifying whole code.

Reply:
Ok so `result[:-1]` will remove last character from resulting string. Just ensure it will be a space.